sklearn/cross_decomposition/pls_.py

```diff
@@ -375,14 +375,14 @@ def transform(self, X, Y=None, copy=True):
         x_scores if Y is not given, (x_scores, y_scores) otherwise.
         """
         check_is_fitted(self, 'x_mean_')
-        X = check_array(X, copy=copy)
+        X = check_array(X, copy=copy, dtype=np.float64)
```

Comment:
Sorry looks like `dtype=FLOAT_DTYPES` prefers `np.float64` over `np.float32`! So we have to do either `as_float_array` or `dtype=[np.float32, np.float64]`!!